packages/eslint-plugin-excel-custom-functions/src/rules/no-office-read-calls.ts

```diff
@@ -0,0 +1,178 @@
+                    const functionStarts = getFunctionStarts2(node, services);
+                    const functionDeclarations = getFunctionDeclarations(node, typeChecker, services);
+                    if (functionDeclarations) {
+                        superNodeMe(functionDeclarations, helperFuncToHelperFuncMap);
+                    }
+
+                    if (functionDeclarations && functionDeclarations.length > 0) {
+                        if(customFunction) {
+                            if (functionDeclarations.some((declaration) => {
+                                return officeCallingFuncs.has(declaration);
+                            })) {
+                                ruleContext.report({
+                                    messageId: "officeReadCall",
+                                    loc: node.loc,
+                                    node: node
+                                });
+                            } else {
+                                let newMentionsArray = helperFuncToMentionsMap.get(functionDeclarations[0]);
+                                helperFuncToMentionsMap.set(functionDeclarations[0], 
+                                    newMentionsArray ? 
+                                    newMentionsArray.concat({
+                                        messageId: "officeReadCall",
+                                        loc: node.loc,
+                                        node: node
+                                    }) :
+                                    [{
+                                        messageId: "officeReadCall",
+                                        loc: node.loc,
+                                        node: node
+                                    }]
+                                );
+                            }
+                        }
+                        
+                        functionStarts.forEach((functionStart) => {
+                            let newHelperFuncSet = helperFuncToHelperFuncMap.get(functionDeclarations[0]);
+                            if (!newHelperFuncSet) {
+                                newHelperFuncSet = new Set<ts.Node>([]);
+                            }
+                            helperFuncToHelperFuncMap.set(functionDeclarations[0], 
+                                newHelperFuncSet.add(functionStart)
+                            );
+                            let bubbledUpSet = bubbleUpNewCallingFuncs(functionStart, helperFuncToHelperFuncMap);
+                            bubbledUpSet.forEach((bubbledUp) => {
+                                let mentions = helperFuncToMentionsMap.get(bubbledUp);
+                                mentions?.forEach((mention) => {
+                                    ruleContext.report(mention);
+                                })
+                                helperFuncToMentionsMap.delete(bubbledUp);
+                            })
+                        });
+                    }
```

Comment:
In general, more comments will help understand the logic for someone new to the code.

Reply:
Also makes doing code reviews easier :-)